src/core/dbus-execute.c

```diff
@@ -980,12 +985,20 @@ static int property_get_exec_dir_symlink(
         if (r < 0)
                 return r;
 
-        FOREACH_ARRAY(i, d->items, d->n_items)
+        FOREACH_ARRAY(i, d->items, d->n_items) {
+                /* The old exec directory properties cannot represent flags, so list them here with no
+                 * destination */
+                if ((i->flags & _EXEC_DIRECTORY_PUBLIC) != 0 && !strv_isempty(i->symlinks)) {
+                        r = sd_bus_message_append(reply, "(sst)", i->path, "", (uint64_t) (i->flags & _EXEC_DIRECTORY_PUBLIC));
+                        if (r < 0)
+                                return r;
+                }
```

Comment:
i fing this a bit hackish, but i guess i am ok wit this. but i think property_get_exec_dir() should always show everything it can, as before, and not "hide" entries just because they have public flags.